File: algorithms/[219]存在重复元素 II/solution.py
# topics = ["哈希表"]

from typing import Dict, List


class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        d: Dict[int, int] = {}

        for i, v in enumerate(nums):
            if v not in d:
                d.setdefault(v, i)
            elif i - d[v] <= k:
                return True
            d[v] = i

        return False

File: algorithms/[219]存在重复元素 II/solution.ts
// topics = ["哈希表"]

function containsNearbyDuplicate(nums: number[], k: number): boolean {
  const map = new Map<number, number>();

  for (const [i, v] of nums.entries()) {
    if (map.has(v) && i - map.get(v)! <= k) {
      return true;
    }
    map.set(v, i);
  }

  return false;
}
